x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/transform/transforms/TransformTaskState.java

```diff
@@ -14,6 +14,7 @@
 import java.util.Locale;
 
 public enum TransformTaskState implements Writeable {
+    // TODO 8.x add a `STOPPING` state and BWC handling in ::fromString
```

Comment:
We could do this earlier than 8.x given the feature is beta.  It's too late for 7.5 but it would probably be worth adding `STOPPING` to 7.6 before making the feature GA.

Reply:
@hendrikmuhs the `state` object will have a new field called `should_stop_at_checkpoint`

Reply:
`TransformStats.State` is what we return to the user, it's a combined value made from the indexer and task state. `TransformState`is only written to/ read from the index, it's not exposed to the user since the stats endpoint has been rewritten. We could also expose a new field, but I think this isn't needed.

Reply:
@hendrikmuhs 

So you want to adjust `TransformStats.State` to say `STOPPING` when `should_stop_at_checkpoint == true`. Or `STOPPING_AT_CHECKPOINT`? Will need to think on this...